static/js/brand-store/Settings/Settings.test.js

```diff
@@ -15,59 +15,105 @@ jest.mock("react-router-dom", () => ({
   useParams: jest.fn().mockReturnValue({ id: "test" }),
 }));
 
-describe("<Settings />", () => {
-  const testState = {
+let initialState = {};
+
+const setupMockSelector = (state) => {
+  useSelector.mockImplementation((callback) => {
+    return callback(state);
+  });
+};
+
+beforeEach(() => {
+  initialState = {
     currentStore: {
       currentStore: {
-        id: "test",
+        id: "test-id",
         private: true,
         "manual-review-policy": "allow",
       },
     },
   };
+});
 
-  beforeEach(() => {
-    useSelector.mockImplementation((callback) => {
-      return callback(testState);
-    });
-  });
+test("the 'is public' checkbox should not be checked when the current store is set to private", () => {
+  setupMockSelector(initialState);
 
-  it("should render the Settings view", () => {
-    render(
-      <Provider store={store}>
-        <Settings />
-      </Provider>
-    );
-    expect(document.querySelector("#is_public")).toBeInTheDocument();
-  });
+  render(
+    <Provider store={store}>
+      <Settings />
+    </Provider>
+  );
 
-  test("the 'is public' checkbox should not be checked when the current store is set to private", () => {
-    render(
-      <Provider store={store}>
-        <Settings />
-      </Provider>
-    );
-    expect(document.querySelector("#is_public").checked).toEqual(false);
-  });
+  expect(document.querySelector("#is_public")).not.toBeChecked();
```

Comment:
Would you consider focusing less on implementation details and making this more user-centric?
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-querying-by-text

```suggestion
  expect(screen.getByLabelText("Include this store in public lists")).not.toBeChecked();
```

Reply:
If you want to be more specific you could use `getByRole`:
```suggestion
  expect(
    screen.getByRole("checkbox", { name: "Include this store in public lists" })
  ).not.toBeChecked();
```

Reply:
I'll do it in this case as that text isn't likely to change anytime soon, but in general I don't think it's a good idea to query by the text as in a lot of cases in our products the text can change a lot which shouldn't break tests (IMO).

When writing tests I keep the following in mind:

> Tests should be sensitive to changes in the behaviour of the code under test. If the behaviour changes, the test result should change.
> Source: https://medium.com/@kentbeck_7670/test-desiderata-94150638a4b3

I don't think text changes fall into this category. Something to discuss at dev catchup 😁